api/v1alpha1/shared_types.go

```diff
@@ -474,6 +474,27 @@ type BackendRef struct {
 	gwapiv1.BackendObjectReference `json:",inline"`
 }
 
+type BackendCluster struct {
```

Comment:
This can be confusing:  In the context of non-route targets(ExtAuth,  ExtProc, OpenTelemetry), `BackendCluser` consists of multiple `BackendRef`s. However, in the context of xRoute, a `BackendRef` itself is a Cluster.